meson.build

```diff
@@ -0,0 +1,1626 @@
+
+
+# checking compiler features
+
+c_compiler = meson.get_compiler('c')
+
+lws_have_malloc_trim = c_compiler.compiles('''
+    #include <malloc.h>
+	int main(int argc, char **argv) { return malloc_trim(0); }
+	''')
+
+lws_have_malloc_usable_size = c_compiler.compiles('''
+	#include <malloc.h>
+	int main(int argc, char **argv) { return (int)malloc_usable_size((void *)0); }
+	''')
+
+
+lws_have_fork = c_compiler.has_function('fork')
+lws_have_getenv = c_compiler.has_function('getenv')
+lws_have_malloc = c_compiler.has_function('malloc')
+lws_have_memset = c_compiler.has_function('memset')
+lws_have_realloc = c_compiler.has_function('realloc')
+lws_have_socket = c_compiler.has_function('(socket')
+lws_have_strerror = c_compiler.has_function('strerror')
+lws_have_vfork = c_compiler.has_function('vfork')
+lws_have_execvpe = c_compiler.has_function('(execvpe')
+lws_have_getifaddrs = c_compiler.has_function('getifaddrs')
+lws_have_snprintf = c_compiler.has_function('snprintf')
+lws_have__snprintf = c_compiler.has_function('_snprintf')
+lws_have__vsnprintf = c_compiler.has_function('_vsnprintf')
+lws_have_getloadavg = c_compiler.has_function('(getloadavg')
+lws_have_atoll = c_compiler.has_function('(atoll')
+lws_have__atoi64 = c_compiler.has_function('_atoi64')
+lws_have__stat32i64 = c_compiler.has_function('_stat32i64')
+lws_have_clock_gettime = c_compiler.has_function('clock_gettime')
+lws_have_eventfd = c_compiler.has_function('eventfd')
+
+if not lws_have_getifaddrs
+	if lws_without_builtin_getifaddrs
+		error('No getifaddrs was found on the system. Turn off the LWS_WITHOUT_BUILTIN_GETIFADDRS compile option to use the supplied BSD version.')
+	endif
+	lws_builtin_getifaddrs = true
+endif
+
+lws_have_dlfcn_h = c_compiler.has_header('dlfcn.h')
+lws_have_fcntl_h = c_compiler.has_header('fcntl.h')
+lws_have_in6addr_h = c_compiler.has_header('in6addr.h')
+lws_have_memory_h = c_compiler.has_header('memory.h')
+lws_have_netinet_in_h = c_compiler.has_header('netinet/in.h')
+lws_have_stdint_h = c_compiler.has_header('stdint.h')
+lws_have_stdlib_h = c_compiler.has_header('stdlib.h')
+lws_have_strings_h = c_compiler.has_header('strings.h')
+lws_have_string_h = c_compiler.has_header('string.h')
+lws_have_sys_prctl_h = c_compiler.has_header('sys/prctl.h')
+lws_have_sys_socket_h = c_compiler.has_header('sys/socket.h')
+lws_have_sys_sockio_h = c_compiler.has_header('sys/sockio.h')
+lws_have_sys_stat_h = c_compiler.has_header('sys/stat.h')
+lws_have_sys_types_h = c_compiler.has_header('sys/types.h')
+lws_have_unistd_h = c_compiler.has_header('unistd.h')
+lws_have_vfork_h = c_compiler.has_header('vfork.h')
+lws_have_sys_capability_h = c_compiler.has_header('sys/capability.h')
+lws_have_malloc_h = c_compiler.has_header('malloc.h')
+lws_have_pthread_h = c_compiler.has_header('pthread.h')
+lws_have_inttypes_h = c_compiler.has_header('inttypes.h')
+
+lws_libcap = c_compiler.find_library('cap', required: false)
+lws_have_libcap = lws_libcap.found()
+if lws_have_libcap
```

Comment:
You don't have to put this conditional here. As far as I know you can unconditionally depend on it, if the library is not found it will be ignored since it's marked with `required: false`.

Reply:
Conditional is removed.
Is this also true for the dependency() function? There are some similar constructs around line 550 (I've more or less converted the 3.000 lines `CMakeLists.txt` from the upstream project into a 1.600 lines `meson.build` file without further thinking).

Reply:
Yes I think so, https://mesonbuild.com/Reference-manual.html#dependency

Reply:
More unnecesessary conditionals are removed.
I like this behavior of `dependency()` function and friends - I did'nt notcied it reading the docs  the first time. Thanks for the advice.